src/ui/dialogs/scenesettingsdialog.cpp

```diff
@@ -308,3 +309,19 @@ void SceneSettingsDialog::sNewSceneDialog(Document& document,
 
     dialog->show();
 }
+
+void SceneSettingsDialog::updateDuration() {
+    const QString typetime = mTypeTime->currentData().toString();
+    const qreal fps = mFPSSpinBox->value();
+    int duration = mMaxFrameSpin->value() - mMinFrameSpin->value();
+
+    if (typetime == "Frames") {
```

Comment:
Use index, as you titles can changed (or be translated).

Reply:
Example:

* Get current index (int) from mTypeTime or add it to the function and let the connection give us the index
* Use a switch statement for index
* Also need to validate the frame range, what if start frame is higher than end frame?

Reply:
What do you suggest to do in that validation? I mean, in case it's not valid:

- error message
- don't change anything
- other

Reply:
Just return, same with the switch statement, default to return.

```
if (min > max) { return; }

switch(index) {
case 0:
duration=...
break;
case 1:
duration=..
break;
default: return;
}
```